mobile_payment/gql_mutations.py

```diff
@@ -0,0 +1,421 @@
+                raise PermissionDenied(_("unauthorized"))
+            client_mutation_id = data.get("client_mutation_id")
+            # it create data inputs from the premium graphql
+            payment_service_provider = update_or_create_payment_service_provider(data, user)
+            PaymentServiceProviderMutation.object_mutated(user, client_mutation_id=client_mutation_id, payment_service_provider=payment_service_provider)
+            return None
+        except Exception as exc:
+            logger.debug("Exception when deleting premium %s", exc_info=exc)
+            return [{
+                'message': _("wallet.mutation.failed_to_create_wallet"),
+                'detail': str(exc)}
+            ]
+
+
+class UpdatePaymentServiceProvider(OpenIMISMutation):
+    _mutation_module = "mobile_payment"
+    _mutation_class = "UpdatePaymentServiceProvider"
+
+
+    class Input (PaymentServiceProviderInputType):
+        pass
+
+    @classmethod
+    def async_mutate(cls, user, **data):
+
+        try:
+            if not user.has_perms(MobilePaymentConfig.gql_mutation_update_paymnet_service_provider):
+                raise PermissionDenied(_("unauthorized"))
+            #get merchant uuid
+            payment_service_provider_uuid = data.pop('uuid')
+            if payment_service_provider_uuid:
+                #fetch merchnat uuid from database
+                payment_service_provider = PaymentServiceProvider.objects.get(uuid = payment_service_provider_uuid)
+                [setattr(payment_service_provider, key, data[key]) for key in data]
+            else:
+                 #raise an error if uuid is not valid or does not exist
+                 raise PermissionDenied(_("unauthorized"))
+            #saves update dta
+            payment_service_provider.save()
+            return None
+        except Exception as exc:
+            return [{
+                
+                'message': _("wallet.mutation.failed_to_update_wallet"),
+                'detail': str(exc)}]
+
+
+class DeletePaymentServiceProvider(OpenIMISMutation):
+
+    _mutation_module = 'mobile_payment'
+    _mutation_class = 'DeletePaymentServiceProvider'
+
+
+    class Input(OpenIMISMutation.Input):
+        uuid = graphene.String()
+
+    @classmethod
+    def async_mutate(cls, user, **data):
+
+        try:
+             # Check if user has permission
+            if not user.has_perms(MobilePaymentConfig.gql_mutation_delete_paymnet_service_provider):
+                raise PermissionDenied(_("unauthorized"))
+            
+             # get programs object by uuid
+            payment_service_provider = PaymentServiceProvider.objects.get(uuid=data['uuid'])
+
+            # get current date time
+            from core import datetime
+            now = datetime.datetime.now()
+
+            # Set validity_to to now to make the record invalid
+            payment_service_provider.validity_to = now
```

Comment:
@tdethier ping

Reply:
Yes, I agree, it's better to use `delete_history()`.

Reply:
okay noted it will be added

Reply:
i do not uderstand how it works , how will i know it is deleted if i add it and trigger the delete mutation

Reply:
This is a soft delete (see [the code in openimis core](https://github.com/openimis/openimis-be-core_py/blob/322538f8bacf9af00be88af127e24eb6fba38c12/core/models.py#L64)). It will set the parameters (such as `validity_to`) to their right values.

I think you should do `payment_service_provider.delete_history()` as I said in my first comment.